File: apps/zxtune-android/zxtune/src/main/java/app/zxtune/analytics/Sink.java
package app.zxtune.analytics;

import android.net.Uri;

import androidx.collection.LongSparseArray;

import java.util.HashMap;

import app.zxtune.core.Player;
import app.zxtune.playback.PlayableItem;

interface Sink {

  void logException(Throwable e);

  // elapsed => tag
  void sendTrace(String id, LongSparseArray<String> points);

  void sendPlayEvent(PlayableItem item, Player player);

  void sendBrowserEvent(Uri path, @Analytics.BrowserAction int action);

  void sendSocialEvent(Uri path, String app, @Analytics.SocialAction int action);

  void sendUiEvent(@Analytics.UiAction int action);

  //! @param scopeSize - uris count for add, selection size else
  //TODO: cleanup
  void sendPlaylistEvent(@Analytics.PlaylistAction int action, int param);

  void sendVfsEvent(String id, String scope, @Analytics.VfsAction int action, long duration);

  void sendNoTracksFoundEvent(Uri uri);

  void sendDbMetrics(String name, long size, HashMap<String, Long> tablesRows);
}

File: apps/zxtune-android/zxtune/src/main/java/app/zxtune/analytics/Sink.kt
package app.zxtune.analytics

import android.net.Uri
import androidx.collection.LongSparseArray
import app.zxtune.core.Player
import app.zxtune.playback.PlayableItem

internal interface Sink {
    fun logException(e: Throwable)

    // elapsed => tag
    fun sendTrace(id: String, points: LongSparseArray<String>)
    fun sendPlayEvent(item: PlayableItem, player: Player)
    fun sendBrowserEvent(path: Uri, action: Analytics.BrowserAction)
    fun sendSocialEvent(path: Uri, app: String, action: Analytics.SocialAction)
    fun sendUiEvent(action: Analytics.UiAction)

    //! @param scopeSize - uris count for add, selection size else
    //TODO: cleanup
    fun sendPlaylistEvent(action: Analytics.PlaylistAction, param: Int)
    fun sendVfsEvent(id: String, scope: String, action: Analytics.VfsAction, duration: Long)
    fun sendNoTracksFoundEvent(uri: Uri)
    fun sendDbMetrics(name: String, size: Long, tablesRows: HashMap<String, Long>)

    fun sendEvent(id: String, vararg arguments: Pair<String, *>)
}
